go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -376,6 +377,69 @@ type managementPortEntry struct {
 	config *managementPortConfig
 }
 
+// getEnvNameFromResourceName gets the device plugin env variable from the device plugin resource name.
+func getEnvNameFromResourceName(resource string) string {
+	res1 := strings.ReplaceAll(resource, ".", "_")
+	res2 := strings.ReplaceAll(res1, "/", "_")
+	return "PCIDEVICE_" + strings.ToUpper(res2)
+}
+
+// getDeviceIdsFromEnv gets the list of device IDs from the device plugin env variable.
+func getDeviceIdsFromEnv(envName string) ([]string, error) {
+	envVar := os.Getenv(envName)
+	if len(envVar) == 0 {
+		return nil, fmt.Errorf("unexpected empty env variable: %s", envName)
+	}
+	deviceIds := strings.Split(envVar, ",")
+	return deviceIds, nil
+}
+
+// handleDevicePluginResources tries to retrieve any device plugin resources passed in via arguments and device plugin env variables.
+func handleDevicePluginResources() error {
+	mgmtPortEnvName := getEnvNameFromResourceName(config.OvnKubeNode.MgmtPortDPResourceName)
+	deviceIds, err := getDeviceIdsFromEnv(mgmtPortEnvName)
+	if err != nil {
+		return err
+	}
+	config.OvnKubeNode.DPResourceDeviceIdsMap = make(map[string][]string)
+	config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName] = deviceIds
+	klog.V(5).Infof("Setting DPResourceDeviceIdsMap for %s using env %s with device IDs %v", config.OvnKubeNode.MgmtPortDPResourceName, mgmtPortEnvName, deviceIds)
+	return nil
+}
+
+// takeDevicePluginInterface takes a device ID from a list of device IDs
+func takeDeviceId(deviceIds []string) (string, []string) {
+	ifsLen := len(deviceIds)
+	ret := (deviceIds)[ifsLen-1]
+	(deviceIds) = (deviceIds)[:ifsLen-1]
+	return ret, deviceIds
+}
+
+// handleDevicePluginInterfaces tries to retrieve any device plugin interfaces to be used by the system such as the management port.
+func handleDevicePluginInterfaces() error {
+	if config.OvnKubeNode.MgmtPortNetdev == "" {
+		var deviceId string
+		if len(config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName]) > 0 {
+			deviceId, config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName] = takeDeviceId(config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName])
```

Comment:
I am preparing for Bypass port. For the case when a user decides to use the same resource pool (same list of VFs) for both MGMT and Bypass port.

Reply:
I hope this makes sense. Please let me know otherwise.

Reply:
Not sure how well it would work. Is there a guarantee that devices in the env.variable sorted?
If not, I could think about situation, when ovnkube-node started using VF0 for mgmt and VF1 for bypass ports, but after pod recreation VF1 and VF0 respectively, or just VF1 for bypass port, because user don't want to configure mgmt port. How messy it will be?

Reply:
I think for regular SR-IOV pods, you aren't guaranteed to get the same VF if the pod restarts. The same is true here. As long as there is no changes to the sriov-policies, the env variable won't change. Is there a technical issue if the VF's purpose changes (e.g. from MGMT port to Bypass Port or vice versa), besides it being a bit messy to keep track of from a developer point of view? The user shouldn't need to worry about which VFs are picked for what.

Reply:
Looking at the code, I don't think we can assume the device list is sorted or contiguous.